cmd/iso_package/file.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+// This code belongs to here:
+// https://blog.depa.do/post/copy-files-and-directories-in-go
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+)
+
+func File(src, dst string) error {
+	var err error
+	var srcfd *os.File
+	var dstfd *os.File
+	var srcinfo os.FileInfo
+
+	if srcfd, err = os.Open(src); err != nil {
+		return err
+	}
+	defer srcfd.Close()
+
+	if dstfd, err = os.Create(dst); err != nil {
+		return err
+	}
+	defer dstfd.Close()
+
+	if _, err = io.Copy(dstfd, srcfd); err != nil {
+		return err
+	}
+	if srcinfo, err = os.Stat(src); err != nil {
+		return err
+	}
+	return os.Chmod(dst, srcinfo.Mode())
+}
+
+// Dir copies a whole directory recursively
+func Dir(src string, dst string) error {
+	var err error
+	var fds []os.FileInfo
+	var srcinfo os.FileInfo
+
+	if srcinfo, err = os.Stat(src); err != nil {
+		return err
+	}
+
+	if err = os.MkdirAll(dst, srcinfo.Mode()); err != nil {
+		return err
+	}
+
+	if fds, err = ioutil.ReadDir(src); err != nil {
+		return err
+	}
+	for _, fd := range fds {
+		srcfp := path.Join(src, fd.Name())
+		dstfp := path.Join(dst, fd.Name())
+
+		if fd.IsDir() {
+			if err = Dir(srcfp, dstfp); err != nil {
```

Comment:
Can this recursion kill us at some point?

Reply:
no, I do not think so

Reply:
I wonder how a link to a folder behaves in this case (or a hard-link). is it considered to reply 'yes' to IsDir or not.
in that case, it might lead to an endless recursion if it points to a parent-dir.